sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -215,6 +215,8 @@ abstract class Star extends LeafExpression with NamedExpression {
 case class UnresolvedStar(target: Option[Seq[String]]) extends Star with Unevaluable {
 
   override def expand(input: LogicalPlan, resolver: Resolver): Seq[NamedExpression] = {
+    // When the table does not have any column, the input LogicalPlan does not have any column
+    if (input.output.isEmpty) return Seq.empty[NamedExpression]
```

Comment:
@cloud-fan The latest changes fail a test case for star expansion in Struct. To resolve a star expansion for Stuct, the whole solution is using two steps when it has a one level qualifier. Try to resolve it as a table qualifier at first. If the returned attributes are empty, then try to resolve it as a struct qualifier.